docs/service-loadbalancer.md

```diff
@@ -0,0 +1,288 @@
+configuration, configuration, in which `kube-proxy` serves the request traffic
+for Services of type LoadBalancer (while `AntreaProxy` handles Service requests
+from Pods).
+- `AntreaProxy` with `proxyAll` enabled - in this case, `AntreaProxy` handles
+all Service traffic, including Services of type LoadBalancer.
+- `AntreaProxy` disabled - `kube-proxy` handles all Service traffic, including
+Services of type LoadBalancer.
+
+#### Create an ExternalIPPool custom resource
+
+Service external IPs are allocated from an ExternalIPPool, which defines a pool
+of external IPs and the set of Nodes that the external IPs can be assigned to.
+To learn more information about ExternalIPPool, please refer to [the Egress
+documentation](egress.md#the-externalippool-resource). The example below
+defines an ExternalIPPool with IP range "10.10.0.2 - 10.10.0.10", and it
+selects the Nodes with label "network-role: ingress-node" to host the external
+IPs:
+
+```yaml
+- apiVersion: crd.antrea.io/v1alpha2
+  kind: ExternalIPPool
+  metadata:
+    name: service-external-ip-pool
+  spec:
+    ipRanges:
+    - start: 10.10.0.2
+      end: 10.10.0.10
+    - cidr: 10.10.1.0/28
+    nodeSelector:
+      matchLabels:
+        network-role: ingress-node
+```
+
+#### Create a Service of type LoadBalancer
+
+For Antrea to manage the externalIP for a Service of type LoadBalancer, the
+Service should be created with the `service.antrea.io/external-ip-pool`
+annotation. For example:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+  annotations:
+    service.antrea.io/external-ip-pool: "service-external-ip-pool"
+spec:
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  type: LoadBalancer
+```
+
+You can also request a particular IP from an ExternalIPPool by setting the IP
+to the `loadBalancerIP` field in the Service spec. If the specified IP is
+available in the ExternalIPPool, Antrea will allocate the IP from the
+ExternalIPPool for the Service. For example:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+  annotations:
+    service.antrea.io/external-ip-pool: "service-external-ip-pool"
+spec:
+  selector:
+    app: MyApp
+  loadBalancerIP: "10.10.0.2"
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  type: LoadBalancer
+```
+
+#### Validate Service external IP
+
+Once Antrea allocates an external IP for a Service of type LoadBalancer, it
+will set the IP to the `loadBalancer.ingress` field in the Service resource
+`status`.  For example:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+  annotations:
+    service.antrea.io/external-ip-pool: "service-external-ip-pool"
+spec:
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  clusterIP: 10.96.0.11
+  type: LoadBalancer
+status:
+  loadBalancer:
+    ingress:
+    - ip: 10.10.0.2
```

Comment:
Antrea will also show a hostname here to indicate which Node holds the external IP.